sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/InstrumentationPolicy.java

```diff
@@ -78,8 +85,8 @@ public HttpResponse processSync(HttpPipelineCallContext context, HttpPipelineNex
         Context span = startSpan(context);
         try (AutoCloseable scope = tracer.makeSpanCurrent(span)) {
             HttpResponse response = next.processSync();
-            endSpan(response, span);
-            return response;
+            updateSpan(response, span);
+            return new TraceableResponse(response, span);
```

Comment:
@alzimmermsft what would you think about 
1. `boolean HttpResponse.isBuffered()` method? It'd allow to optimize stuff here and in HttpResponseLogger
2. something like `HttpResponse.addCloseHook(BiSupplier<HttpResponse, Throwable> hook)` - this would help us avoid wrapping streams

Not strictly related, but I think `BinaryData` should also have `isBuffered` or `isBufferable` and `close` to optimize/allow more control to close streams under (e.g. when I don't want to  read full content)

Reply:
I have no issue with these additions. Though I think it'd be good to have a larger discussion around the entire HTTP request-response lifecycle, hooks into it, and metadata associated with it. There are a lot of areas we could improve through these enhancements and it'd be good to get them all in one place so we can be consistent with what we're doing.

Reply:
created https://github.com/Azure/azure-sdk-for-java/issues/38228 to track it